### PR TITLE
[java][cdp] Use devtools script pinning mechanism by default

### PR DESCRIPTION
**Pull request by pujagani**
This helps achieve feature parity in script pinning among different Selenium language bindings.

**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

The script pinning methods as part of the driver did not use the DevTools mechanism of script pinning. The changes fix this by checking DevTools support and using the script pinning commands of DevTools. 

### Motivation and Context

This help achieve parity among languages. The same approach is implemented in C#, Ruby, and JS.
In future, this can be extended to use BiDi's script module command for script pinning.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/remote/RemoteWebDriver.java [hunk 1]
```diff
@@ -493,6 +497,76 @@ public Object executeAsyncScript(String script, Object... args) {
...
+  @Override
+  public void unpin(ScriptKey scriptKey) {
+    UnpinnedScriptKey key = (UnpinnedScriptKey) scriptKey;
+
+    JavascriptExecutor.super.unpin(key);
+
+    String browserName = getCapabilities().getBrowserName().toLowerCase();
+    if ((browserName.equals("chrome") ||
+         browserName.equals("msedge") ||
+         browserName.equals("microsoftedge")) && this instanceof HasDevTools) {
```
Comment (diemol) [inline java/src/org/openqa/selenium/remote/RemoteWebDriver.java:513]: Why do we need to check the browser name? If the driver `HasDevTools` it should be enough, right?

Comment (pujagani) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:513]: Some of the CDP commands are not implemented in Firefox and that causes the functionality to break. 
Hence, the browser check.

Comment (diemol) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:513]: 👍 
The plan is to move Firefox to BiDi as soon as possible?

Comment (pujagani) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:513]: The plan is to add BiDi APIs that are implemented by browsers and support existing Selenium BiDi APIs as soon as we can.

**Comment by brodev3** [comment]
hello, I am delighted with your work, thank you so much for the opportunities provided with your library, please tell me, are you planning to add this function to the Selenium version on python?

**Review by diemol — APPROVED** [review]

**Pull request merged**